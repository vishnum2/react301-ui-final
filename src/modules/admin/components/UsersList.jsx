import React,{useState, useEffect} from 'react';
import axiosWrapper from '../../../apis/axiosCreate';
import { Column, Table, SortDirection, AutoSizer } from "react-virtualized";
import "react-virtualized/styles.css";

const UsersList = () => {

    const [allusers, setAllUsers] = useState(null);
    const [width, setWidth] = useState(window.innerWidth);
    const [height, setHeight] = useState(window.innerHeight* 80 / 100);

    const getusers = async () => {
        console.log('get users running');
        const response = await axiosWrapper.get(`users`);
        setAllUsers(response.data);
    }

    const windowResizeSetter = () => {
        const width = window.innerWidth;
        const height = window.innerHeight*80 / 100;
        setHeight(height);
        setWidth(width);
        // return {width, height};
    }

    useEffect(() => {
        getusers();
        window.addEventListener('resize',windowResizeSetter)
    },[])

    if(window.sessionStorage.getItem('usertype') === 'admin') {
        if(allusers) {
            {console.log(allusers)}
            // const width = window.innerWidth;
            // const height = window.innerHeight*80 / 100;
            // const { width,height } = windowResizeSetter();
            console.log(width, height);
            return (
                <div>
                {/* {console.log(allusers)} */}
                {/* <h1>USERS LIST</h1> */}
            {/* <AutoSizer>
                {() => ( */}
                <Table
                  width={width}
                  height={height}
                  headerHeight={30}
                  rowHeight={30}
                //   sort={this._sort}
                //   sortBy={this.state.sortBy}
                //   sortDirection={this.state.sortDirection}
                  rowCount={allusers.length}
                  rowGetter={({ index }) => allusers[index]}
                >
                  <Column label="Name" dataKey="first_name" width={width/3} />
                  <Column width={width/3} label="Email" dataKey="email" />
                  <Column width={width/3} label="Balance Points" dataKey="balance_points" />
                </Table>
               {/* )}
            </AutoSizer>  */}
                </div>
            )
        } else {
            return (
                <h1> Loading ... </h1>
            )
        }
    } else {
        return (
            <h1>You are not an Admin</h1> 
        )
    }
}

export default UsersList;