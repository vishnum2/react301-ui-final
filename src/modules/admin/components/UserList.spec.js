import React from 'react';
import {shallow, mount} from '../../../enzyme';
import UsersList from './UsersList';
import axiosWrapper from '../../../apis/axiosCreate';

describe('check the snapshot', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<UsersList />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should get all users', () => {
        // window.sessionStorage.getItem = jest.fn().mockImplementationOnce(() => 'admin');
        axiosWrapper.get = jest.fn().mockImplementationOnce(() => Promise.resolve({data: 'new'}));
        window.sessionStorage.setItem('usertype', 'admin');
        const wrapper = mount(<UsersList />);
        expect(axiosWrapper.get).toHaveBeenCalledTimes(1);
    })
})