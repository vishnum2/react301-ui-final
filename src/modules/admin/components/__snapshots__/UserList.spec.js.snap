// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the snapshot should match snapshot 1`] = `ShallowWrapper {}`;
